charts/metering-ansible-operator/templates/crds/hive.crd.yaml

```diff
@@ -21,3 +21,253 @@ spec:
+            external:
+              type: boolean
+              description: |
+                External controls whether an external table is created, instead of a managed table.
+                If external is set to true, this causes Hive to point to an existing location,
+                as specified by the "location" field.
+                Note: this is an optional field.
+            managePartitions:
+              type: boolean
+              description: |
+                ManagePartitions controls whether the reporting-operator needs to check
+                if the table partitions match the partitions listed in the "partitions" field.
+                Note: this is an optional field.
+            partitions:
+              type: array
+              description: |
+                A list of partitions that this Hive table should contain.
+                Note: this is an optional field.
+              nullable: true
+              items:
+                type: object
+                required:
+                - partitionSpec
+                - location
+                properties:
+                  partitionSpec:
+                    type: array
+                    description: |
+                      PartitionSpec is a map containing string keys and values, where each key
+                      is expected to be the name of a partition column, and the value is the
+                      value of the partition column.
+                  location:
+                    type: string
+                    description: |
+                      Location specifies where the data for this partition is stored.
+                      This should be a sub-directory of the "location" field.
+                    minLength: 1
+                    format: uri
+          anyOf:
+          - required:
+            - numBuckets
+            - clusteredBy
+            properties:
+              external:
+                enum:
+                - true
+              managePartitions:
+                enum:
+                - true
+              partitions:
+                type: array
+          - required:
+            - external
+            - location
+            properties:
+              external:
+                enum:
+                - true
+              managePartitions:
+                enum:
+                - true
+              partitions:
+                type: array
+          - required:
+            - managePartitions
+            - partitions
+            properties:
+              managePartitions:
+                enum:
+                - true
+              partitions:
+                type: array
+              external:
+                enum:
+                - true
+          # note: allow the minimal configuration, and enforce the schemas defined above
+          - allOf:
+            - not:
```

Comment:
Maybe im confused but doesn't this prevent setting these fields at all?

Reply:
Yes, this would just allow two configurations, the one where only `tableName`, `databaseName`, and `columns` are provided (the minimal configuration) and also the one that the `PrometheusMetricsImporter` provides

Reply:
Oh, I'd rather we go to the oneOf validating the main requirements than preventing valid configs from being created. That wasn't super clear that supporting the "main" meant "not supporting advanced". I'd rather simply not validate more advanced options than not allow them at all.

Reply:
Unless you mean the nested `allOf` clause preventing the setting of these properties within the context of the overall `anyOf`, which isn't the case. 

This just acts as the default case for the configurations described above.

Reply:
Oh i totally missed this was nested! Shoot.